### DISCUSSION TITLE
Image Optimization issue

### DISCUSSION THREAD

**Discussion by X3Technology** [vanjarosoftware/Vanjaro.Platform#1697] [General]
I am working with this .jpg
![homepage-pco-facein2-hero-fg](https://user-images.githubusercontent.com/1585717/131423306-7c332f3e-88cf-472a-aa61-ddea27ae953b.jpg)

When I upload via the Asset Manager (or the Image block), it optimizes the image into the various formats, which is great.  
But, the webp image that is being displayed is now too pxielated.

GitHub won't let me upload the optimized .webp file, but here is a screenshot of it.
![screen](https://user-images.githubusercontent.com/1585717/131423595-034e78c5-c8ac-4b4a-ad61-75a2ccf9eb36.PNG)

Is this a compression problem?  Not sure how to work around this.

**Comment by meetmandeep** [comment]
The two images are not of same dimensions. Can you provide the original image you're uploading?

**Comment by X3Technology** [comment]
The top one is the original .jpg.  Try uploading it via the Image picker and selecting it.  It comes out pixelates like the second image. It's especially noticeable in the embedded text (I know, bad idea, but not my image).

**Comment by nikhilmandeeps** [comment]
Upload top image via the Asset Manager. The original **.jpg** file is same as below **.webp** image

![131423306-7c332f3e-88cf-472a-aa61-ddea27ae953b](https://user-images.githubusercontent.com/62682589/131620234-a93a4212-8430-4b03-b787-2dc4943c5dba.jpg)

**Reply by X3Technology** [reply]
The image @nikhilmandeeps uploaded and referenced as "below .webp image" is not .webp, it is .jpg.  GitHub doesn't allow you to upload .webp images.

**Reply by nikhilmandeeps** [reply]
@mathisjay uploaded image is screen shot of .webp image

**Reply by X3Technology** [reply]
I didn't make up the .webp images in the attached .zip file.  These came directly from Vanjaro.  The image degradation in the image files that Vanjaro generated is very clear.  I'm not sure how else to explain it...

**Comment by X3Technology** [comment]
Here is a .zip file with the original image (.jpg) and the 4 optimized images (2 .webp and 2 .jpeg).  

[homepage-pco-facein2-hero-fg.zip](https://github.com/vanjarosoftware/Vanjaro.Platform/files/7090794/homepage-pco-facein2-hero-fg.zip)

If you look closely, you will see significant pixilation on the 4 optimized images.  The quality is too low for me to be able to use these.

**Comment by meetmandeep** [comment]
It looks like we need to introduce the "Quality" setting to save the new JPEG/WEBP files. It's always a compromise between performance and image quality. But if we can expose that setting then admin can tweak it as necessary.

**Comment by X3Technology** [comment]
I agree.  It would be ideal if it could be done differently on each individual image.  The default compression setting is probably fine for most images.  

Images with fine detail like person's faces (headshots) or any type of smooth color gradient will have the same issue.

**Comment by X3Technology** [comment]
@nikhilmandeeps 
Here is another test.  I tested using two images.  

1. A 600px x437px  .jpg that contains a person face and some embedded text (test.face.jpg).
2. A 1920px x1200px purple gradient used as a background. (bg.jpg)

I uploaded both image via the Asset Manager.
For image 1, Vanjaro created 4 new optimized images and placed them in the .version folder - 2 600w versions (.webp and .jpg) and 2 360w versions.
For image 2, Vanajro created 8 optimized images (2 360w, 2 720w, 2 1280w, 2 1920w).

I zipped up my .versions folder and the original images in this .zip file to show the results.

[images.zip](https://github.com/vanjarosoftware/Vanjaro.Platform/files/7098862/images.zip)

Please inspect the images and you should be able to see the image degradation very clearly.

For image 1, the degradation is most obvious around the embedded text and details around the persons face and some other gradient effects.  

Here is the original:
![test face](https://user-images.githubusercontent.com/1585717/131846503-374341ed-6775-410e-9ae3-58ce9e6937ce.jpg)

Here is the optimized 600w version with some pointers added to show the problem:
![test face_optimized](https://user-images.githubusercontent.com/1585717/131846355-f284f2d2-e3b5-4168-9aac-b1833b297e47.jpg)

For image 2, the smooth gradient of the original image become very banded, even in the 1920w  versions.  
Here is the original:
![bg](https://user-images.githubusercontent.com/1585717/131846711-0e274d74-da4b-462a-a737-1e9ee3a352bf.jpg)

Here is the optimized 1920w version with some pointers added to show the problem:
![bg_optimized](https://user-images.githubusercontent.com/1585717/131845169-9a5f12af-36db-4cd8-b283-623105763eb3.jpg)

The degradation is enough that the client complained, thus unacceptable.  I can't explain it anymore clearly than this.
Let me know if you have any questions.

**Reply by nikhilmandeeps** [reply]
I am able to duplicate the issue with second image and will be fixed in next update.
